ground/gcs/src/plugins/usagestatsgadget/usagestatsplugin.cpp

```diff
@@ -0,0 +1,315 @@
+            }
+            else {
+                disconnect(slider, SIGNAL(valueChanged(int)), this, SLOT(onSliderValueChanged(int)));
+            }
+        }
+        else if (tab) {
+            if(conn) {
+                connect(tab, SIGNAL(currentChanged(int)), this, SLOT(onTabCurrentChanged(int)), Qt::UniqueConnection);
+            }
+            else {
+                disconnect(tab, SIGNAL(currentChanged(int)), this, SLOT(onTabCurrentChanged(int)));
+            }
+        }
+        else {
+            searchForWidgets(obj, conn);
+        }
+    }
+}
+
+void UsageStatsPlugin::onButtonClicked()
+{
+    widgetActionInfoType info;
+    QAbstractButton *button = qobject_cast<QAbstractButton*>(sender());
+    info.objectName = button->objectName();
+    info.data1 = button->text();
+    info.data2 = QString::number(button->isChecked());
+    info.className = button->metaObject()->className();
+    info.parentName = button->parent()->objectName();
+    info.time = QDateTime::currentDateTime();
+    info.type = WIDGET_BUTTON;
+    widgetLogList.append(info);
+}
+
+void UsageStatsPlugin::onSliderValueChanged(int value)
+{
+    widgetActionInfo info;
+    QAbstractSlider *slider = qobject_cast<QAbstractSlider*>(sender());
+    info.objectName = slider->objectName();
+    info.data1 = QString::number(value);
+    info.className = slider->metaObject()->className();
+    info.parentName = slider->parent()->objectName();
+    info.time = QDateTime::currentDateTime();
+    info.type = WIDGET_SLIDER;
+    widgetLogList.append(info);
+}
+
+void UsageStatsPlugin::onTabCurrentChanged(int value)
+{
+    widgetActionInfo info;
+    QTabBar *tab = qobject_cast<QTabBar*>(sender());
+    info.objectName = tab->objectName();
+    info.data2 = QString::number(value);
+    info.data1 = tab->tabText(value);
+    info.className = tab->metaObject()->className();
+    info.parentName = tab->parent()->objectName();
+    info.time = QDateTime::currentDateTime();
+    info.type = WIDGET_TAB;
+    widgetLogList.append(info);
+}
+
+QByteArray UsageStatsPlugin::processJson(QString ip){
+    QJsonObject json;
+    if(!ip.isEmpty())
+        json["IP"] = ip;
+    else
+        json["IP"] = "PRIVATE";
+    json["gcs_version"] =  QLatin1String(Core::Constants::GCS_VERSION_LONG);
+    json["gcs_revision"] =  QLatin1String(Core::Constants::GCS_REVISION_PRETTY_STR);
+    QJsonArray boardArray;
+    foreach (boardLog board, boardLogList) {
+        QJsonObject b;
+        b["time"] = board.time.toString();
+        b["name"] = board.board.board.data()->getBoardNameFromID(board.device.boardID());
+        b["uavoHash"] = QString(board.device.uavoHash.toHex());
+        b["ID"] = board.device.boardID();
+        b["fwHash"] = QString(board.device.fwHash.toHex());
+        b["gitDate"] = board.device.gitDate;
+        b["gitHash"] = board.device.gitHash;
+        b["gitTag"] = board.device.gitTag;
+        if(sendPrivateData)
+            b["CPU"] = board.board.cpu_serial;

```

Comment:
Can we send a hash (e.g. SHA-512) of the serial number instead? In the future, I plan to offer the possibility of people registering their brainfpv flight controllers and the serial number will be required for this, so making the serial public may allow people to register using the serial number from someone else. Using a hash would give the same functionality, i.e., we could count the number of unique boards etc.